modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -512,11 +594,24 @@ def map_func(left, right=right, kwargs=kwargs):
     def reindex(self, axis, labels, **kwargs):
         new_index = self.index if axis else labels
         new_columns = labels if axis else self.columns
+        actual_index = None
+        if "_reset_index" in kwargs:
+            actual_index = kwargs.pop("_reset_index")
+
+        def _reindex(df):
+            df = df.reindex(labels=labels, axis=axis, **kwargs)
```

Comment:
pandas.DataFrame.reindex can take copy=False, can we do that here?

Reply:
let's try